bee-message/src/error.rs

```diff
@@ -121,6 +126,7 @@ pub enum Error {
     MissingPayload,
     MissingRequiredSenderBlock,
     MissingStateControllerUnlockCondition,
+    MissingStorageDepositReturnUnlockCondition,
```

Comment:
That can't be an error, either you have one or not,  but since it's optional it can't be missing

Reply:
Hmm, what about the case where the output amount is to small, but no `StorageDepositReturnUnlockCondition` is posted?

Reply:
I think it would be nice to differentiate between these two cases though.

Reply:
I think it should be `InvalidStorageDepositReturnAmount`, otherwise you would always get `StorageDepositReturnUnlockCondition` as error unless you have one? I think in most output you don't expect that unlock condition

Reply:
If introduced two new variants `InsufficientStorageDepositAmount` and `InsufficientStorageDepositReturnAmount` because `InvalidStorageDepositAmount` is used by `Packable` to denote out-of-bounds errors. I'm not settled on the names yet, so if you have suggestions... 😉